src/dune_rules/setup.mli

```diff
@@ -1,21 +1,12 @@
 (** Setup of dune *)
 open! Dune_engine
 
-(** These parameters are set by [ocaml configure.ml] or copied from
-    [setup.defaults.ml]. *)
+(** These parameters are set by [ocaml configure.ml] ran by the user or without
+    argument as a fallback. *)
 
-(** Where to find installed libraries for the default context. If [None],
+(** Where to find installed libraries for the default context. If empty,
     auto-detect it using standard tools such as [ocamlfind]. *)
-val library_path : string list option
+val library_path : string list
 
-(** Where to install libraries for the default context. *)
-val library_destdir : string option
-
-(** Where to install manpages for the default context. *)
-val mandir : string option
-
-(** Where to install docs for the default context. *)
-val docdir : string option
-
-(** Where to install configuration files for the default context. *)
-val etcdir : string option
+(** Where to install files. All the directories are absolute path *)
+val roots : string option Install.Section.Paths.Roots.t
```

Comment:
I don't like this change to setup. Before it was 100% readable/writeable for end users, now they need to look up some record definition just so it saves the devs a bit of boilerplate. Seems like the wrong trade-off to me.

Reply:
I thought we always supported writing `setup.ml` by hand as a configuration mechanism. I checked the docs and we indeed only mention `configure.ml` so I suppose this change is technically fine.